types/react/index.d.ts

```diff
@@ -557,23 +557,140 @@ declare namespace React {
     // Class Interfaces
     // ----------------------------------------------------------------------
 
+    /**
+     * Represents the type of a function component. Can optionally
+     * receive a type argument that represents the props the component
+     * receives.
+     *
+     * @param P - The props the component receives.
+     * @see {@link https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components}
```

Comment:
:speech_balloon: It would be nice if this was at all covered in the actual React docs and we could link to that.

I actually assumed that `React.FC` was mentioned in https://react.dev/learn/typescript , but it isn't. That just shows types for `props` (inline or extracted).